test/integration/minikube_profile_test.go

```diff
@@ -0,0 +1,53 @@
+package integration
+
+import (
+	"context"
+	"encoding/json"
+	"os/exec"
+	"testing"
+
+	"k8s.io/minikube/pkg/minikube/config"
+)
+
+// profileJSON is the output of `minikube profile list -ojson`
+type profileJSON struct {
+	Valid   []config.Profile `json:"valid"`
+	Invalid []config.Profile `json:"invalid"`
+}
+
+func TestMinikubeProfile(t *testing.T) {
+	// 1. Setup two minikube cluster profiles
+	profiles := []string{UniqueProfileName("first"), UniqueProfileName("second")}
+	ctx, cancel := context.WithTimeout(context.Background(), Minutes(10))
+	// TODO(@vharsh): Figure out why go vet complains when this is moved into a loop
+	defer CleanupWithLogs(t, profiles[0], cancel)
+	defer CleanupWithLogs(t, profiles[1], cancel)
+	for _, p := range profiles {
+		_, err := Run(t, exec.CommandContext(ctx, Target(), "start", "-p", p))
+		if err != nil {
+			t.Fatalf("test pre-condition failed: minikube start -p " + p + " failed: " + err.Error())
+		}
+	}
+	// 2. Change minikube profile
+	for _, p := range profiles {
+		_, err := Run(t, exec.CommandContext(ctx, Target(), "profile", p))
+		if err != nil {
+			t.Fatalf("minikube profile %s failed with error: %v\n", p, err.Error())
+		}
+		r, err := Run(t, exec.CommandContext(ctx, Target(), "profile", "list", "-ojson"))
+		if err != nil {
+			t.Fatalf("minikube profile list -ojson failed with error: %v\n", err.Error())
+		}
+		var profile profileJSON
+		err = json.NewDecoder(r.Stdout).Decode(&profile)
+		if err != nil {
+			t.Fatalf("minikube profile list -ojson failed with error: %v\n", err.Error())
+		}
+		// 3. Assert minikube profile is set to the correct profile in JSON
+		for _, s := range profile.Valid {
+			if s.Name == p && !s.Active {
+				t.Fatalf("minikube profile %s is not active\n", p)
```

Comment:
```suggestion
				t.Errorf("minikube profile %s is not active\n", p)
```
This one is an outcome vs a setup, so it's fine to make this an `Error` instead of a `Fatal`. And makes more sense with the above comment, that way we can loop through all the profiles and see all of the failures instead of just quitting on the first one.